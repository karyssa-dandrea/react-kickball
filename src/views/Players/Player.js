import React from 'react';
import { useState, useEffect } from 'react';
import { fetchPlayerById } from '../../services/players';
import PlayerDetail from './Players';
export default function Player(props) {
  const [player, setPlayer] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchPlayerById(id);
      setPlayer(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) {
    return <h1>Loading</h1>;
  }

  return (
    <div className="player-deets">
      <h1>Players</h1>
      <p>{player.name}</p>
      <PlayerDetail player={player} />
    </div>
  );
}
