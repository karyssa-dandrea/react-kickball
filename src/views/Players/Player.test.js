import Player from '../../views/Players/Player';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test('should render correct player when clicked', async () => {
  const { container } = render(
    <MemoryRouter>
      <Player match={{ params: { id: 8 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Buster Rimes');
  expect(container).toMatchSnapshot();
});
