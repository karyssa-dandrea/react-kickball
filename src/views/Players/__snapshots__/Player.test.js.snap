// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct player when clicked 1`] = `
<div>
  <div
    class="player-deets"
  >
    <p>
      Buster Rimes
    </p>
    <div
      class="player-details"
    >
      <p>
        Position: 
        Shortstop
      </p>
    </div>
  </div>
</div>
`;
